modules/ROOT/pages/clauses/load-csv.adoc

```diff
@@ -397,4 +570,367 @@ If `LOAD CSV` is invoked with a `file:///` URL that points to your disk `file()`
+- *Special or newline characters* -- When dealing with special characters in a file, ensure they are quoted or remove them.
+- *Inconsistent line breaks* -- Ensure line breaks are consistent throughout your file.
+We recommend the Unix style for compatibility with Linux systems. (and is compatible with windows? does it even matter, is not up to the jvm to do the parsing, which is system agnostic?)
+- *Binary zeros, BOM byte order mark and other non-text characters* -- Unusual characters or tool-specific formatting are sometimes hidden in application tools, but become apparent in plain-text editors.
+If you come across these types of characters in your file, remove them entirely.
+
+
+==== Inspect source files ahead of import
+
+Before actually importing data into the database, you can use `LOAD CSV` to inspect a source file and get an idea of what form the imported data is going to have.
+
+.Inspect CSV files with `LOAD CSV`
+====
+[source, cypher]
+----
+// Assert correct line count
+LOAD CSV FROM 'https://data.neo4j.com/importing-cypher/persons.csv' AS line
+RETURN count(*);
+
+// Check first 5 line-sample with header-mapping
+LOAD CSV WITH HEADERS FROM 'https://data.neo4j.com/importing-cypher/persons.csv' AS line
+RETURN line
+LIMIT 5;
+----
+
+.Result
+[source, role="queryresult"]
+----
++-------------------+
+| No data returned. |
++-------------------+
+Nodes created: 4
+Properties set: 8
+Labels added: 4
+----
+
+====
+
+== Performance recommendations
+
+
+=== Create and use indexes
+
+xref:indexes/index.adoc[Indexes] can vastly speed up queries targeting the indexes entities.
+Either before or after the import, create indexes on the labels and properties that you expect to query most often.
+
+.Create indexes on movie ID and title
+[source, cypher]
+----
+CREATE INDEX movie_imdbId FOR (m:Movie) ON (m.movie_imdbId) IF NOT EXISTS;
+CREATE INDEX movie_title FOR (m:Movie) ON (m.title) IF NOT EXISTS;
+----
+
+For more information on the role of indexes, see xref:indexes/search-performance-indexes/using-indexes.adoc[].
+
+
+=== Separate creation of nodes and relationships
+
+It is more efficient to create _all_ nodes first, and then add relationships with a second pass.
+
+.Comparison: import nodes and relationships together VS splitting them
+====
+.https://data.neo4j.com/importing-cypher/books.csv
+[source, csv]
+----
+id,title,author,publication_year,genre,rating,still_in_print,last_purchased
+19515,The Heights,Anne Conrad,2012,Comedy,5,true,2023/4/12 8:17:00
+39913,Starship Ghost,Michael Tyler,1985,Science Fiction|Horror,4.2,false,2022/01/16 17:15:56
+60980,The Death Proxy,Tim Brown,2002,Horror,2.1,true,2023/11/26 8:34:26
+18793,Chocolate Timeline,Mary R. Robb,1924,Romance,3.5,false,2022/9/17 14:23:45
+67162,Stories of Three,Eleanor Link,2022,Romance|Comedy,2,true,2023/03/12 16:01:23
+25987,Route Down Below,Tim Brown,2006,Horror,4.1,true,2023/09/24 15:34:18
+----
+
+.Bad practice -- Create nodes and relationships at once
```

Comment:
This is pretty harsh. A user may not have enough data to import to warrant splitting it up, there must be a threshold where this is fine to do. Also there must be overhead with setting up the connection etc. Same with running two separate queries - there is going to be a point where the amount of data versus setting up a db connection has an effect on the "badness" of splitting